src/engine/handler/join-practice.js

```diff
@@ -0,0 +1,62 @@
+const Player = require('../game/player')
+const { singleTargetMessage, multiTargetMessage } = require('../../message/socket-message')
+
+class JoinPracticeHandler {
+	constructor(games, messagesToSocketStream$) {
+		this.games = games;
+		this.messagesToSocketStream$ = messagesToSocketStream$
+	}
+
+	handle(message) {
+		const game = this.games.newPractice()
+			.on('countdown-finished', () => {
```

Comment:
We need to think about this, how to handle those "on" conditions so they are easier to understand or even share